File: python/InsertionSort/src/InsertionSort.py
# Python program for implementation of Insertion Sort
import sys

# main
list = map(int, sys.argv[1:len(sys.argv)])
print "Input list::", list

# Function to do insertion sort
def insertionSort(list):

    # Traverse through 1 to len(list)
    for i in range(1, len(list)):

        print
        insertionValue = list[i]

        print "iteration::", i, "::InsertionSort::", insertionValue

        position = i

        # Move elements of list[0..i-1], that are
        # greater than insertionValue, to one position ahead
        # of their current position
        while position > 0 and list[position-1] > insertionValue :
            print "swap::[", list[position -1], " --> ", list[position], "] position::[", position, "]"

            #swap
            list[position] = list[position-1]
            position -= 1

        print "swap::[", list[position], " <-- ", insertionValue, "] position::[", position, "]"

        #swap Insertion Value to correct position
        list[position] = insertionValue

        print "Partial list::", list

# Driver code to test above
insertionSort(list)
print "Output list::", list

File: java/InsertionSort/src/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {

  public static void main(String[] list) {

    int position;
    String insertValue;

    System.out.println("Input::".concat(Arrays.toString(list)));

    for(int i = 1; i < list.length; i++) {

      position = i;
      insertValue = list[i];

      System.out.println("iteration::" + i);
      System.out.println("insertValue::" + insertValue);

      while(position > 0 && Integer.parseInt(list[position -1]) > Integer.parseInt(insertValue)) {

        System.out.println("swap::[" + list[position -1] + " --> " + list[position] + "]");

        list[position] = list[position -1];

        position--;
      }

      System.out.println("swap::[" + list[position] + " <-- " + insertValue + "]");
      list[position] = insertValue;

      System.out.println("Partial::".concat(Arrays.toString(list)));
    }

    System.out.println("Output::".concat(Arrays.toString(list)));
  }
}
